rules/anemic-vs-rich-domain-models/rule.md

```diff
@@ -0,0 +1,113 @@
+---
+seoDescription: Understand the differences between anemic and rich domain models to improve your software architecture.
+type: rule
+title: Do you understand the difference between anemic and rich domain models?
+uri: anemic-vs-rich-domain-models
+authors:
+  - title: Daniel Mackay
+    url: https://www.ssw.com.au/people/daniel-mackay
+created: 2024-06-17T13:42:55.753Z
+guid: 346475FF-7255-4BE2-BD09-C8113CA40399
+---
+
+When using Domain-Centric architectures such as Clean Architecture, we need to decide where the business logic will go. There are two main approaches to this: Anemic Domain Model and Rich Domain Model. Understanding the differences between these two models is crucial for making informed decisions about your software architecture.
+
+<!--endintro-->
+
+## Anemic Domain Model
+
+An Anemic Domain Model is characterized by:
+
+- **Property Bags:** Entities are simple data containers with public getters and setters.
+- **No Behavior:** No logic or validation within the entities.
+
+### Pros of Anemic Domain Model:
+
+- Good for simple or CRUD projects.
+- Easier to understand for developers new to the project.
+
+### Cons of Anemic Domain Model:
+
+- Doesn’t scale well with complexity.
+- Logic duplication across the application.
+- Difficult to maintain as the project grows.
+
+::: greybox
+class Order {
+    public int Id { get; set; }
+    public DateTime OrderDate { get; set; }
+    public decimal TotalAmount { get; set; }
+}
+:::
+::: bad
```

Comment:
This isn't a bad example, it's just an example. The rule isn't "Do you understand that rich domain models are better than anemic domain models?"

I would just show it as an example, and get rid of the "bad", but if you want good/bad examples, you probably need good and bad examples for both. E.g.:

## Anemic:

* **Bad example:** Repeated logic (more complex than simple CRUD) for the same entity in multiple places throughout the application
* **Good example**: Simple order class (like you've got), end point with CRUD operations (and no logic).


## Rich

* **Bad example:** All of the code in the domain model even though it's just doing simple CRUD (arguably nothing wrong with this actually so would need to be qualified)
* **Good example**: Logic in domain model (more complex than simple CRUD) reused across the Application.

Reply:
You're right these are just "examples".  Have removed good and bad.  Haven't specifically shown good and bad examples.